temporal-sdk/src/main/java/io/temporal/client/WorkflowStub.java

```diff
@@ -145,22 +145,50 @@ <R> R update(
    * @param firstExecutionRunId specifies the RunID expected to identify the first run in the
    *     workflow execution chain. If this expectation does not match then the server will reject
    *     the update request with an error.
+   * @param waitPolicy specifies at what point in the update request life cycles this request should
+   *     return.
    * @param resultClass class of the update return value.
    * @param <R> type of the update return value.
-   * @param resultType type of the workflow return value. Differs from resultClass for generic
-   *     types.
+   * @param resultType type of the update return value. Differs from resultClass for generic types.
    * @param args update method arguments
-   * @return update reference that can be used to get the result of the update.
+   * @return update handle that can be used to get the result of the update.
    */
   @Experimental
   <R> UpdateHandle<R> startUpdate(
       String updateName,
       String updateId,
       String firstExecutionRunId,
+      UpdateWorkflowExecutionLifecycleStage waitPolicy,
```

Comment:
I'm a bit torn on whether to wrap this in a friendlier enum. I know Java doesn't in some other places, but this name is nothing like the parameter name. I guess it can stay as is, it's just very ugly.

Reply:
I +1 that. This name is a mouthful